python/sqlflow_submitter/xgboost/dataset.py

```diff
@@ -79,21 +81,28 @@ def thread_worker(slice_id):
         threads.append(t)
     map(lambda t: t.join(), threads)
 
-
-def pai_download_table_data_worker(dir_or_file_name, feature_specs,
-                                   feature_column_names, label_spec, pai_table,
-                                   slice_id, single_file):
+    if single_file:
+        import glob
+        filenames = glob.glob('%s/*.txt' % dname)
+        with open(filename, 'w') as fw:
+            for fname in filenames:
+                with open(fname) as fr:
+                    for line in fr:
+                        fw.write(line)
```

Comment:
Maybe can use `cat file1.txt file2.txt file3.txt file4.txt > ./mergedfile.txt` command which maybe fast because of the linux redirect is default buffered.